source/solvers/navier_stokes_assemblers.cc

```diff
@@ -1712,16 +1712,16 @@ BuoyancyAssembly<dim>::assemble_rhs(
       // Current temperature values
       double current_temperature = scratch_data.temperature_values[q];
 
-      strong_residual[q] += -force * thermal_expansion * current_temperature;
+      strong_residual[q] += force * thermal_expansion * current_temperature;
 
       // Assembly of the right-hand side
       for (unsigned int i = 0; i < n_dofs; ++i)
         {
           const auto phi_u_i = scratch_data.phi_u[q][i];
 
           // Laplacian on the velocity terms
-          local_rhs(i) +=
-            -force * thermal_expansion * current_temperature * phi_u_i * JxW;
+          local_rhs(i) -=
```

Comment:
same comment as above.
Clearly, if you are not sure about the signs, take the time to add the information to the theory guide and it will help you sort it out. it's there for that.